gmso/utils/misc.py

```diff
@@ -5,6 +5,19 @@
 from unyt.exceptions import UnitConversionError
 
 
+@lru_cache(maxsize=128)
+def unyt_compare(units1, units2):
+    """Check if two parameter values have the same units."""
+    units1 = list(units1)
+    units2 = list(units2)
+    for unit1, unit2 in zip(units1, units2):
+        try:
+            u.testing.assert_allclose_units(unit1, unit2, rtol=1e-5, atol=1e-8)
+            return True
+        except AssertionError:
+            return False
```

Comment:
The testing.assert_allclose_units function raises an Assertion error itself if unit1 and unit2 are not within tolerance. So that should fail the try statement and go to the except and return False, which will then fail the assertion I laid out in the main __eq__ check. There's probably a better way around this, I'm open to thoughts. I just didn't want this check to raise an assertion error everytime you check equality, it should be return True or False.